File: Android/DustAPI_UsingJava/app/src/main/java/com/example/dustapi_usingjava/MainActivity.java
package com.example.dustapi_usingjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.res.Resources;
import android.net.Uri;
import android.os.AsyncTask;
import android.os.Bundle;
import android.os.StrictMode;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.TextView;

import org.w3c.dom.Text;
import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserFactory;

import java.net.URL;
import java.util.ArrayList;
import java.util.List;


public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    private Button Get_But;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        StrictMode.enableDefaults();

        Resources res = getResources();
        String[] items = res.getStringArray(R.array.location_array);
        ArrayList<String> list = new ArrayList<String>();

        for (String it:items){
            list.add(it);
        }
        ArrayAdapter myAdapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, list);

        Spinner loc_spinner = (Spinner)findViewById(R.id.location_spinner);
        loc_spinner.setAdapter(myAdapter);

        Get_But = (Button)findViewById(R.id.Get_Button);
        Get_But.setOnClickListener(this);
    }

    public void onClick(View view){
        TextView Dust_Txt = (TextView)findViewById(R.id.Dust_Text);
        GetDustInformation gd = new GetDustInformation();
        try {
            URL url = new URL("http://openapi.airkorea.or.kr");
            gd.execute(url);
        }
        catch (Exception E){
            Dust_Txt.setText("hphoho");
        }
    }

    class GetDustInformation extends AsyncTask<URL, Integer, Long>{
        @Override
        protected void onPreExecute(){

        }

        @Override
        protected Long doInBackground(URL... urls){
            TextView Dust_Txt = (TextView)findViewById(R.id.Dust_Text);
            try {
                Spinner loc_spinner = (Spinner)findViewById(R.id.location_spinner);
                URL url = new URL("http://openapi.airkorea.or.kr/openapi/services/rest/ArpltnInforInqireSvc/getMsrstnAcctoRltmMesureDnsty?stationName="
                        + loc_spinner.getSelectedItem().toString() +
                        "&dataTerm=month&pageNo=1&numOfRows=10&ServiceKey=V6i2Rt1FL4jsYItn2EEcQm6Aw6AB4C0WsWt9YbSXem%2FiCQjiMGQnfVzrGUtkbKZY0dA7EJawhznH%2FEb6wtHowg%3D%3D&ver=1.3");
                System.out.println(url);
                XmlPullParserFactory parserCreator = XmlPullParserFactory.newInstance();
                XmlPullParser parser = parserCreator.newPullParser();
                parser.setInput(url.openStream(), null);

                int parserEvent = parser.getEventType();
                System.out.println("파싱시작합니다.");
                Boolean pm10=false, pm25=false, end=false;
                Integer pm10val=0, pm25val=0;
                while (parserEvent != XmlPullParser.END_DOCUMENT) {
                    switch (parserEvent) {
                        case XmlPullParser.START_TAG://parser가 시작 태그를 만나면 실행
                            System.out.print(parser.getName());
                            if (parser.getName().equals("pm10Value")) pm10=true;
                            if (parser.getName().equals("pm25Value")) pm25=true;
                            break;

                        case XmlPullParser.TEXT://parser가 내용에 접근했을때
                            System.out.print(parser.getText());
                            if (pm10){
                                System.out.println(parser.getText());
                                System.out.println("여기가문제인감33");
                                pm10val = Integer.parseInt(parser.getText());
                                System.out.println("요쪽이문제인감55");
                            }
                            else if (pm25){
                                System.out.println(parser.getText());
                                System.out.println("여기가문제인감");
                                pm25val = Integer.parseInt(parser.getText());
                                System.out.println("요쪽이문제인감");
                                end=true;
                            }
                            break;
                        case XmlPullParser.END_TAG:
                            if (parser.getName().equals("pm10Value")) pm10=false;
                            if (parser.getName().equals("pm25Value")) pm25=false;
                            break;
                    }
                    if (end) break;
                    parserEvent = parser.next();
                }
                Dust_Txt.setText("미세먼지 농도 : " + pm10val + ",\n\n초미세먼지 농도 : " + pm25val);
            }
            catch(Exception E){
                Dust_Txt.setText("인터넷 연결 오류!");
            }

            return 0L;
        }

        @Override
        protected void onProgressUpdate(Integer... progress){

        }

        @Override
        protected void onPostExecute(Long result){

        }
    }
}

File: Android/Dust_APIConnect/app/src/main/java/com/example/dust_apiconnect/MainActivity.kt
package com.example.dust_apiconnect

import android.content.Context
import android.content.Context.CONNECTIVITY_SERVICE
import android.net.ConnectivityManager
import android.net.NetworkInfo
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*
import java.net.URL
import java.net.URLConnection
import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.*
import javax.net.ssl.HttpsURLConnection

fun isConnected(context: Context):Boolean{
    val manager = context.getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
    val networkInfo: NetworkInfo? = manager.activeNetworkInfo?:null

    if (networkInfo != null){
        return true
    }
    return false
}

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val items = resources.getStringArray(R.array.my_array)
        val myAdapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, items)
        spinner.adapter = myAdapter

        if (isConnected(this)) {
            textView2.setText("Internet Connected")
        }
        else{
            textView2.setText("Internet Not Connected")
        }
        button.setOnClickListener{
            var url = URL(
                "http",
                "openapi",
                80,
                "/openapi/services/rest/ArpltnInforInqireSvc/getMsrstnAcctoRltmMesureDnsty?stationName=" + spinner.getSelectedItem() + "&dataTerm=month&pageNo=1&numOfRows=10&ServiceKey=V6i2Rt1FL4jsYItn2EEcQm6Aw6AB4C0WsWt9YbSXem%2FiCQjiMGQnfVzrGUtkbKZY0dA7EJawhznH%2FEb6wtHowg%3D%3D&ver=1.3"
            )
            val connection = url.openConnection()

            try {
                var item=false
                var end=false;
                var pm25Val=false
                var pm10Val=false
//,"/openapi/services/rest/ArpltnInforInqireSvc/getMsrstnAcctoRltmMesureDnsty?stationName="+spinner.getSelectedItem()+"&dataTerm=month&pageNo=1&numOfRows=10&ServiceKey=V6i2Rt1FL4jsYItn2EEcQm6Aw6AB4C0WsWt9YbSXem%2FiCQjiMGQnfVzrGUtkbKZY0dA7EJawhznH%2FEb6wtHowg%3D%3D&ver=1.3"

                System.out.println("")
                System.out.println("protocol name:" + url.getProtocol());
                System.out.println("host name:" + url.getHost());
                System.out.println("file name:" + url.getFile());
                System.out.println("port name:" + url.getPort());


                val `is` = connection.inputStream
                val sb = StringBuilder()
                val br = BufferedReader(InputStreamReader(`is`, "UTF-8"))
                var result: String



/*                while (parserEvent != XmlPullParser.END_DOCUMENT) {
                    when (parserEvent) {
                        XmlPullParser.START_TAG -> {
                            if (parser.getName().equals("item")) {
                                item = true
                            }
                            if (parser.getName().equals("/item")) {
                                end = true
                            }
                            if (parser.getName().equals("pm10Value")) {
                                pm10Val = true
                            }
                            if (parser.getName().equals("pm25Value")) {
                                pm25Val = true
                            }
                        }
                        XmlPullParser.TEXT -> {
                            if (pm10Val) {
                                textView2.setText("미세먼지 값 : " + parser.getText())
                                pm10Val = false
                            }
                            if (pm25Val) {
                                textView3.setText("초미세먼지 값 : " + parser.getText())
                                pm25Val = false
                            }
                        }
                    }
                    if (end) break;
                    parserEvent = parser.next();
                }*/
            }
            catch(e:NumberFormatException){
                textView2.setText("에러가 났습니다...!!")
            }
        }
    }
}
